### PR TITLE
Bl 870 fe i want to create a new course

### PR DESCRIPTION
**Pull request by macrowell70**
## Description

For this ticket we created a form for administrators to add new courses. The form was created using AntDesign for both the form itself and the modal it is housed in. Eventually this will be used for editing courses in addition to adding new courses, so the form has been designed to be updated for editing information in the future.

contributors: Mitchel Crowell & Lisa Despain

Fixes # (issue)

## Loom Video

https://www.loom.com/share/bfc0d2636d4544008b08c96af3a5ea2b

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] I have made corresponding changes to the documentation if necessary (optional)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

File: src/components/pages/AdminAddCourse/AdminAddCoursesForm.js [hunk 1]
```diff
@@ -1,181 +1,320 @@
...
+              </Form.Item>
+              <Form.Item
+                label={
+                  <label style={{ color: '#096A70', fontSize: '1.1rem' }}>
+                    Program:
+                  </label>
+                }
+                style={{ width: '75%', color: '#096A70', fontSize: '1.1rem' }}
               >
-                Clear
```
Comment (ashtilawat23) [inline src/components/pages/AdminAddCourse/AdminAddCoursesForm.js:160]: Love that we removed all this excess code!

**Review by ashtilawat23 — DISMISSED** [review]
Great work on this PR! 

Some questions I have: 
- could we move inline styles to a designated file? 
- could we use `.map` in repeated patterns in files
- address the unused variable in your PR

Awesome work!

File: src/components/pages/AdminAddCourse/AdminAddCoursesForm.js [hunk 2]
```diff
@@ -1,181 +1,320 @@
...
+                  style={{ fontSize: '1.1rem' }}
+                />
+              </Form.Item>
+              <Form.Item
+                label={
+                  <label style={{ color: '#096A70', fontSize: '1.1rem' }}>
+                    Course Description:
+                  </label>
+                }
+                style={{ width: '75%', color: '#096A70', fontSize: '1.1rem' }}
```
[the remark this answers is not in the document]
Comment (lisamdespain) [reply src/components/pages/AdminAddCourse/AdminAddCoursesForm.js:151]: We could remove the color from inline (it's actually addressed in the CSS file too) but I couldn't get the font size to do ANYTHING in css, even after borrowing code from another component that did fix the size. The only thing that seemed to work at all was inline. Would love to have someone else take a look and/or experiment with it to see if it's possible.

[the remark this answers is not in the document]
Comment (ashtilawat23) [reply src/components/pages/AdminAddCourse/AdminAddCoursesForm.js:151]: Ok no problem. I can try to take a look today, but for now lets leave them here. Maybe we can make a bug ticket as well to transfer them later.

**Review by reneetro — DISMISSED** [review]
Awesome work! Good job on formatting the date and time inputs, that will help will validation for sure. Clean code!

File: src/components/pages/AdminAddCourse/AdminAddCoursesForm.js [hunk 3]
```diff
@@ -1,181 +1,320 @@
...
 import TextArea from 'antd/lib/input/TextArea';
-import { useHistory } from 'react-router-dom';
-
-let placeHolder = [];
-let array_string = '';
-let program_list = [];
-let formPrerequisite = '';
+const { Option } = Select;
 
 const initialFormValues = {
```
[the remark this answers is not in the document]
Comment (macrowell70) [reply src/components/pages/AdminAddCourse/AdminAddCoursesForm.js]: I've moved the initial form values to AdminAddCoursesPage.js

File: src/components/pages/AdminAddCourse/AdminAddCoursesPage.js [hunk 4]
```diff
@@ -1,16 +1,55 @@
-import React from 'react';
+import React, { useState } from 'react';
 
 import { connect } from 'react-redux';
+import { useOktaAuth } from '@okta/okta-react';
+import { useDispatch } from 'react-redux';
+import { addCourse, editCourse } from '../../../redux/actions/coursesActions';
 import '../../../styles/AdminAddCoursesStyles/AdminAddCoursesPage.less';
```
[the remark this answers is not in the document]
Comment (macrowell70) [reply src/components/pages/AdminAddCourse/AdminAddCoursesPage.js:7]: Removed editCourses

[the remark this answers is not in the document]
Comment (macrowell70) [reply src/components/pages/AdminAddCourse/AdminAddCoursesPage.js:7]: Also, I've mapped the number fields in the form to clean that up a little.

**Review by theecolya — APPROVED** [review]
It looks pretty good, great base code for me to turn into a universal modal component

**Review by Roadlyfe — DISMISSED** [review]
this looks great!

**Review by CaitlinTalbert — COMMENTED** [review]
Code looks clean and easy to follow! Great video walking through what you guys did.

**Review by lailaarkadan — APPROVED** [review]
Great job walking through and thoroughly explaining your code. Liked how you added the field where you can select individual days and the option to select the start and end times.

**Review by DawsonReschke — DISMISSED** [review]
Great work, code looks clean.

**Pull request merged**